rlp/hash.go

```diff
@@ -10,26 +10,38 @@ import (
 
 // Hash returns the keccak256 hash of the encoded RLP value as a hexadecimal string prefixed with 0x
 func (v Value) Hash() (string, error) {
+
+	sum, err := v.HashToBytes()
+	if err != nil {
+		return "", err
+	}
+
+	// and finally return the hash as a 0x prefixed string
+	digest := hex.EncodeToString(sum)
+	return "0x" + digest, nil
+}
+
+// Hash returns the keccak256 hash of the encoded RLP in []byte
```

Comment:
```suggestion
// HashToBytes returns the keccak256 hash of the encoded RLP as a byte slice
```

Reply:
I'm not 100% convinced this function is necessary, but it's hard to say at the moment because of the mixing of strings and coerced byte slices.  I think it'd be cleaner and generally more in line with the rest of this module to keep everything as hexadecimal strings.

Reply:
I guess I wasn't sure what the advantage of using hexadecimal string is..  It seems like we remove the `0x` by using [:2] and then decode it back from a string in the `Recover` function.. which seems rather inefficient to add + encode only to remove + decode..